conformance/lang/expressions/unary-expression/unary_minus_expression.balt

```diff
@@ -0,0 +1,1079 @@
+
+const float A = -10f;
+const float B = -20f;
+
+type Floats1 -2.0|-1.0|0|1.0|2.0;
+
+type Floats2 A|B;
+
+type Floats3 Floats1|Floats2;
+
+function init() {
+    Floats2? a = -10f;
+    Floats3? b = -1.0;
+
+    io:println(-a); // @output 10.0
+    io:println(-b); // @output 1.0
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive optional decimals, no overflow scenarios.
+Labels: DecimalFloatingPointNumber, decimal, optional-type, nil-literal, numeric-nil-lifting, unary-minus, value:toBalString
+
+function init() {
+    decimal? a = 1.0;
+    decimal? b = 101.5;
+    decimal? c = ();
+    decimal? d = 922439289.0;
+    io:println(-a); // @output -1.0
+    io:println(-b); // @output -101.5
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output -922439289.0
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive optional user defined sub types of decimals,
+              no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-lang#32928
+Labels: DecimalFloatingPointNumber, module-type-defn, singleton-type, unary-minus, union-type, numeric-nil-lifting, optional-type, FloatingPointTypeSuffix
+
+type Decimals -2.0d|-1.0d|0d|1.0d|2.0d;
+
+function init() {
+    Decimals? a = 1.0;
+    Decimals? b = 2.0;
+    Decimals? c = 0;
+
+    io:println(-a); // @output -1.0
+    io:println(-b); // @output -2.0
+    io:println(-c); // @output 0
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive user defined sub types of decimals,
+              no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-lang#32928
+Labels: DecimalFloatingPointNumber, decimal, module-type-defn, module-const-decl, unary-minus, union-type, singleton-type, numeric-nil-lifting, optional-type, FloatingPointTypeSuffix
+
+const decimal A = -10d;
+const decimal B = -20d;
+
+type Decimals1 -2.0d|-1.0d|0d|1.0d|2.0d;
+
+type Decimals2 A|B;
+
+type Decimals3 Decimals1|Decimals2;
+
+function init() {
+    Decimals2? a = -10;
+    Decimals3? b = -1.0;
+
+    io:println(-a); // @output 10
+    io:println(-b); // @output 1.0
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative optional decimals, no overflow scenarios.
+Labels: DecimalFloatingPointNumber, decimal, optional-type, nil-literal, numeric-nil-lifting, unary-minus, value:toBalString
+
+function init() {
+    decimal? a = -1.0;
+    decimal? b = -101.5;
+    decimal? c = ();
+    decimal? d = -922439289.0;
+    io:println(-a); // @output 1.0
+    io:println(-b); // @output 101.5
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output 922439289.0
+}
+
+Test-Case: output
+Description: Test field and member access as unary negation operands.
+Labels: array-type, DecimalNumber, int, list-constructor, mapping-constructor, module-type-defn, record-type, unary-minus
```

Comment:
missing `member-access-expr`. let's add in other places as well if there is any usage.

Reply:
`field-access-expr` is also missing